spec/models/principal_spec.rb

```diff
@@ -203,6 +203,25 @@
     end
   end
 
+  describe '#onboarded?' do
+    context 'when a principal has been onboarded' do
+      before :each do
+        principal.firm.update_column(:email_address, 'acme@example.com')
```

Comment:
Plus, simply setting the email address doesn't really create an onboarded firm. Is there something missing in the logic that decides what onboarded means?

Maybe for this test what we really should do is mock `next_onboarding_action` to return known states.